### ISSUE TITLE
Value null at 'createStageInput.deploymentId'

### ISSUE THREAD

**Issue by timoteialbu** [svdgraaf/serverless-plugin-stage-variables#15]
Hey,

I was following this [article](https://forum.serverless.com/t/knowledge-sharing-enable-cloud-watch-logs-for-api-gateway-using-serverless/3287) which suggested using your plugin, but when I try deploying I get the following error: 
`An error occurred: ApiGatewayStage - 1 validation error detected: Value null at 'createStageInput.deploymentId' failed to satisfy constraint: Member must not benull (Service: AmazonApiGateway; Status Code: 400; Error Code: ValidationException; Request ID: b4a97ac8-9f17-11e8-ad43-2f6b195fc4e8).` 

Any thoughts? Why is that variable not exported? Do you think its an issue with the plugin or on serverless side?

Thanks for your time,

Timotei

**Comment by svdgraaf** [comment]
I think that's a recent change in CF, see also #14 whcih I think is the same issue, I'll look into it.

**Comment by svdgraaf** [comment]
Just noticed, that's actually you refering to it ;)

**Comment by svdgraaf** [comment]
Can you post the output of the `AWS::ApiGateway::Stage` resource in your cloudformation file in the `.serverless` directory? It should have the deployment Id filled in, but could be something changed in a recent version of Serverless.

**Comment by timoteialbu** [comment]
Thanks for getting back to me so quick!

This is what I have in the `cloudformation-template-update-stack.json`:
```
"ApiGatewayStage": {
      "Type": "AWS::ApiGateway::Stage",
      "Properties": {
        "RestApiId": {
          "Ref": "ApiGatewayRestApi"
        },
        "MethodSettings": [
          {
            "DataTraceEnabled": true,
            "HttpMethod": "*",
            "LoggingLevel": "INFO",
            "ResourcePath": "/*",
            "MetricsEnabled": true
          }
        ]
      }
    }
```

**Comment by markforcepoint** [comment]
As a workaround I used the serverless-plugin-bind-deployment-id (https://github.com/jacob-meacham/serverless-plugin-bind-deployment-id) To insert the deploymentId

```  ApiGatewayStage:
      Type: 'AWS::ApiGateway::Stage'
      Properties:
        DeploymentId:
          Ref: __deployment__
        RestApiId:
          Ref: ApiGatewayRestApi
        StageName: ${self:provider.stage}
        MethodSettings:
         - DataTraceEnabled: true
           HttpMethod: "*"
           LoggingLevel: INFO
           ResourcePath: "/*"
           MetricsEnabled: true
```

**Comment by johnnyplaydrums** [comment]
Hey, any movement on this issue? Just ran into it as well

**Comment by svdgraaf** [comment]
I don't have any time currently to work on this, some help (or PR) is greatly appreciated...

**Comment by johnnyplaydrums** [comment]
Cool no worries, I’ll try to carve out some time for it. Thanks!

**Comment by chrisyorkston** [comment]
Just hit this issue, is there any update? Thanks

**Comment by dsmileym4** [comment]
just hit this issue today as well.

**Comment by mdgreenwald** [comment]
Also encountering this issue.

**Comment by dmhalejr** [comment]
Hey I think I know what the issue is here. I'd like to claim this issue.

**Comment by svdgraaf** [comment]
I just released `1.8.1` with @dmhalejr fix in it. Please check if this resolves your issue.

**Comment by svdgraaf** [comment]
I just released 1.8.1 with @dmhalejr's bugfix in it. Please see if this fixes your issues.

**Comment by robinweston** [comment]
I'm afraid that @dmhalejr 's fix doesn't work. It actually makes things a bit worse. `TestDeployment` is just a dummy value used in an AWS Guide. The accepted PR hardcodes the DeploymentId to this value so there is no hope of matching up to the correct (random) deployment Id. I am investigating why the correct Deployment Id isn't picked up as the code looks sound

**Comment by robinweston** [comment]
Ok, so I've got to the bottom of it.

@dmhalejr 's PR has no effect as the DeploymentId is immediately overwritten by `stageConfig.Properties.DeploymentId = {"Ref":key}`

The null DeploymentId was happening for me simply because the plugin hook never fired. This is fixed by [this PR](https://github.com/svdgraaf/serverless-plugin-stage-variables/commit/b4ee6d401857daf57445888aa11ca83b5c789af7#diff-1fdf421c05c1140f6d71444ea2b27638) and just needs publishing as a new version to npm

**Comment by jamesallardice** [comment]
@svdgraaf Any chance you could publish the latest changes to npm? As per the last comment on this issue, there's a change in master which resolves an issue but is not published. It's easy to work around by installing the package directly from GitHub but it would be preferable to use the registry with proper semver.

**Comment by svdgraaf** [comment]
@jamesallardice oh crap, did I not publish it? on it!

**Comment by svdgraaf** [comment]
I have pushed version 1.8.2 to npm!

**Comment by jamesallardice** [comment]
🎉 thanks for the quick turnaround @svdgraaf!

**Comment by limaneto** [comment]
Still facing this issue, is this problem still around?

**Comment by svdgraaf** [comment]
@limaneto afaik this should be resolved by the bump to 1.8.2

**Comment by justinlazaro-iselect** [comment]
having an issue here, same error. any fix? 

```
ApiGatewayStage:
    Type: AWS::ApiGateway::Stage
    Properties:
      RestApiId:
        Ref: ApiGatewayRestApi
      MethodSettings:
        - DataTraceEnabled: true
          HttpMethod: '*'
          LoggingLevel: INFO
          ResourcePath: '/*'
          MetricsEnabled: true
```
